security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -166,7 +170,11 @@ func (s *sdsservice) StreamSecrets(stream sds.SecretDiscoveryService_StreamSecre
 				ResourceName: resourceName,
 			}
 			addConn(key, con)
-			defer removeConn(key)
+			defer func() {
+				removeConn(key)
+				// Remove the secret from cache.
```

Comment:
It would be more informational to comment why we do this instead of what this is doing :)

Maybe something like "To avoid duplicated notifications of same secret expiry"

Reply:
added more comments